src/login/logind-button.c

```diff
@@ -181,6 +217,19 @@ static int button_dispatch(sd_event_source *s, int fd, uint32_t revents, void *u
                         break;
                 }
 
+        } else if (ev.type == EV_KEY && ev.value == 0) {
+                if (ev.code == KEY_RESTART) {
+                        log_struct(LOG_INFO,
```

Comment:
please add some explanatory comment here, that this is about key release, and why we check for the timer event source here. i doubt this would be obvious for the casual onlooker